src/main/java/org/skriptlang/skript/addon/AddonModule.java

```diff
@@ -44,4 +52,43 @@ default boolean canLoad(SkriptAddon addon) {
 		return true;
 	}
 
+	/**
+	 * Returns the documentation category which this module belongs to.
+	 *
+	 * @return The category of this module.
+	 */
+	default @NotNull Set<Category> category() {
+		return Set.of();
+	}
+
+	/**
+	 * Registers syntax such that it belongs to the current module.
+	 *
+	 * @param addon The addon this module belongs to.
+	 * @param registry The registry to add this syntax to.
+	 * @param cls The syntax info.
+	 * @param <I> The type of syntax.
+	 */
+	default <I extends SyntaxInfo<?>> void register(SkriptAddon addon, Key<I> registry, I cls) {
+		//noinspection unchecked
+		addon.syntaxRegistry().register(registry, (I) cls.toBuilder()
+				.origin(SyntaxOrigin.of(addon, this))
+				.build());
+	}
+
+	/**
+	 * Registers syntax such that it belongs to the current module.
+	 *
+	 * @param addon The addon this module belongs to.
+	 * @param registry The registry to add this syntax to.
+	 * @param classes The syntax infos.
+	 * @param <I> The type of syntax.
+	 */
+	@SuppressWarnings("unchecked")
+	default <I extends SyntaxInfo<?>> void register(SkriptAddon addon, Key<I> registry, I... classes) {
+		for (I cls : classes) {
+			register(addon, registry, cls);
```

Comment:
It would be better to have a private method that takes in the SyntaxRegistry instead. It's not that `SkriptAddon#syntaxRegistry` is super expensive, but it may not be as simple as returning a field.

Reply:
it's an interface though, so it wouldn't be private

Reply:
the more important part is getting the registry outside of the loop.